clang/lib/CIR/CodeGen/CIRGenBuilder.h

```diff
@@ -139,8 +139,9 @@ class CIRGenBuilderTy : public cir::CIRBaseBuilderTy {
   }
 
   bool isSized(mlir::Type ty) {
-    if (mlir::isa<cir::PointerType, cir::ArrayType, cir::BoolType,
-                  cir::IntType>(ty))
+    if (mlir::isa<cir::PointerType, cir::ArrayType, cir::BoolType, cir::IntType,
+                  cir::CIRFPTypeInterface, cir::ComplexType, cir::RecordType>(
```

Comment:
It would be nice if we figured out a way to make patches that add new types 'check' here.  a `switch` on the `TypeID` is typically what we'd suggest in the FE, but of course that doesn't work here for a number of reasons.  Would be nice if someone thought of a way to make sure we didn't forget this again.

Reply:
What about creating `IsSized` type trait that is attached to type definition, so one does not need to track uses like this?

Reply:
This should do it https://github.com/llvm/clangir/pull/1714